packages/cosmos_utils/lib/amount_formatter.dart

```diff
@@ -0,0 +1,22 @@
+import 'package:intl/intl.dart';
+
+String formatAmount(double amount) {
+  final formatted = NumberFormat.currency(symbol: '\$').format(amount);
+  final raw = formatted.split('.')[0].substring(1).replaceAll(',', '');
+  if (raw.length >= 7) {
+    if (raw.length >= 16) {
+      return _getShortAmount(raw, 15, 'Q'); // Quadrillion
+    }
+    if (raw.length >= 13) {
+      return _getShortAmount(raw, 12, 'T'); // Trillion
+    }
+    if (raw.length >= 10) {
+      return _getShortAmount(raw, 9, 'B'); // Billion
+    }
+    return _getShortAmount(raw, 6, 'M'); // Million
+  }
+  return formatted;
```

Comment:
@andrzejchm 
Came up with this idea of adding `M`, `B`, `T`, and so on, for formatting amount along with simple formatting. Do let me know if we can think of something better than this.

Reply:
how about we use the `NumberFormat.compact()` formatter?

https://stackoverflow.com/a/54691084/4838100

Reply:
That's perfect for our use case!

Reply:
Updated ✅